### PR TITLE
refactor/581 AddLoan component

### LINKED ISSUE #581: Refactor AddLoanComponent

### PR DESCRIPTION
**Pull request by te4-henrik-stahl**
## ⚙️ What issue did you implement or fix?
Fixes #581

## 📜 Summary
- Refactored AddLoan Component

## 📝 Type of change
- [x] ✨ New featureor enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by te4-dennis-christensen — APPROVED** [review]
Looks good

File: frontend/src/components/AddLoan.vue [hunk 1]
```diff
@@ -61,91 +61,94 @@
...
-import { Component, Watch, Prop, Vue } from 'vue-property-decorator';
 import LoansModule from '../store/modules/LoansModule';
 import { Loan, AddLoanForm } from '../types';
 import UsersModule from '../store/modules/UsersModule';
+import { defineComponent, ref, SetupContext, watch } from '@vue/composition-api';
+export default defineComponent({
+  name: 'AddLoan',
+  setup(_ : object, { root, emit } : SetupContext ) {
+    
+    let inputText = ref(''),
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/AddLoan.vue:72]:
```suggestion
    const inputText = ref(''),
```

File: frontend/src/components/AddLoan.vue [hunk 2]
```diff
@@ -61,91 +61,94 @@
...
+      inputState = ref(false),
+      inputReturn = ref(''),
+      button = ref(null),
+      input = ref(''),
+      menu = ref(false as boolean);
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/AddLoan.vue:77]: Is the `as boolean` required?

File: frontend/src/components/AddLoan.vue [hunk 3]
```diff
@@ -61,91 +61,94 @@
...
-    lent_by_id: 1854282603, //eslint-disable-line camelcase
-    loaned_by_id: 0,//eslint-disable-line camelcase
-    expiration_date: '2020/02/12' //eslint-disable-line camelcase
-  };
-
+    const form: AddLoanForm  = {
+      book_id: 0, //eslint-disable-line camelcase
+      /* Hardcoded value atm for the school */
+      lent_by_id: 1854282603, //eslint-disable-line camelcase
+      loaned_by_id: 0,//eslint-disable-line camelcase
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/AddLoan.vue:83]:
```suggestion
      loaned_by_id: 0, //eslint-disable-line camelcase
```

File: frontend/src/components/AddLoan.vue (hunk 1 above)
Comment (itggot-linus-styren) [inline frontend/src/components/AddLoan.vue:72]: Prefer to use const here

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Looks good. There may be some tweaks that can be done. See my suggestions.

File: frontend/src/components/AddLoan.vue (hunk 2 above)
Comment (itggot-linus-styren) [inline frontend/src/components/AddLoan.vue:77]: "as boolean" is redundant, type is inferred by literal value.

File: frontend/src/components/AddLoan.vue [hunk 4]
```diff
@@ -61,91 +61,94 @@
...
+      if(!inputText.value){
+        inputState.value = false;
+      }
+      if (!input.value) {
+        onCancel();
+        /* "Return" our popover "form" results */
+        inputReturn.value = input.value;
+        form.book_id = parseInt(input.value); //eslint-disable-line camelcase
+        form.loaned_by_id = UsersModule.currentUserID; //eslint-disable-line camelcase
+        console.log(form);
```
Comment (itggot-linus-styren) [inline frontend/src/components/AddLoan.vue:101]: Can be removed

File: frontend/src/components/AddLoan.vue [hunk 5]
```diff
@@ -61,91 +61,94 @@
...
+        /* This Checks if the form contains the necessary data  */
+        if (!!form.lent_by_id && !!form.loaned_by_id && !!form.book_id) {
```
Comment (itggot-linus-styren) [inline frontend/src/components/AddLoan.vue:103]:
```suggestion
        if (!Object.values(form.value).some(prop => prop === 0)) {
```
Somewhat more clean solution.

**Review by itggot-linus-styren — COMMENTED** [review]
Some comments, also added original author of component as reviewer

**Review by te4-andre-skvarc — CHANGES_REQUESTED** [review]
See what the other reviewers said and correct it!

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Looks good. But please also check the previous comments and the linter errors. 
https://github.com/itggot-TE4/Yabs/pull/590#discussion_r546675896

File: frontend/src/components/AddLoan.vue [hunk 6]
```diff
@@ -61,91 +61,93 @@
...
+        inputState.value = false;
+      }
+      if (!input.value) {
+        onCancel();
+        /* "Return" our popover "form" results */
+        inputReturn.value = input.value;
+        form.book_id = parseInt(input.value); //eslint-disable-line camelcase
+        form.loaned_by_id = UsersModule.currentUserID; //eslint-disable-line camelcase
+        /* This Checks if the form contains the necessary data  */
+        if (!Object.values(form.value).some(prop => prop === 0)) {
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/AddLoan.vue:102]:
```suggestion
        if (!Object.values(form).some(prop => prop === 0)) {
```

File: frontend/src/components/AddLoan.vue (hunk 2 above)
Comment (te4-henrik-stahl) [reply frontend/src/components/AddLoan.vue:77]: Im not 100% sure but i remember something about us needing it, but i could be wrong.

Comment (te4-adrian-almetun-smeds) [reply frontend/src/components/AddLoan.vue:77]: Shouldn't be needed though..

File: frontend/src/components/AddLoan.vue [hunk 7]
```diff
@@ -76,13 +76,14 @@ export default defineComponent({
       input = ref(''),
       menu = ref(false as boolean);
 
-    const form: AddLoanForm  = {
+    const form: Ref < AddLoanForm > = ref ({
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/AddLoan.vue:79]: Didn't know you could do Ref as a type :+1:

**Review by te4-adrian-almetun-smeds — APPROVED** [review]
Looks good to me. :+1: :shipit:

File: frontend/src/components/AddLoan.vue (hunk 2 above)
Comment (te4-henrik-stahl) [reply frontend/src/components/AddLoan.vue:77]: So should just change it to (false) or (null)?

**Review by te4-andre-skvarc — APPROVED** [review]
Good work Henrik!

**Pull request merged**